### DISCUSSION TITLE
Segmentation fault in a gadget in release mode

### DISCUSSION THREAD

**Discussion by achataigner** [gadgetron/gadgetron#1240] [Q&A]
Hello,

I build & run gadgetron 4.1 on a ubuntu 20.04 gcc 9.4.0
I tried to dereference a nullptr in a simple gadget and I got these results on my ubuntu 20.04 pc:
- release mode: no crash, separated threads are killed, main thread is up, I can start new streams and it works
- debug mode: SIGSEGV is raised 

I read many posts and I know nullptr dereferencing leads to "undefined behavior".
But I'm wondering if you are surprised by the result in release mode ?
Is it expected ? It looks like a feature...

Best regards

**Comment by hansenms** [comment]
I think it is because the Gadgetron forks in release mode, so it kills the forked process, but the main process stays up.
